handlers/user.js

```diff
@@ -0,0 +1,598 @@
+const express = require('express');
+const { user: userController } = require("../controllers");
+const validate = require('../validate');
+const { body } = require('express-validator');
+const config = require('../config')
+const jwt = require('jsonwebtoken');
+const user = express.Router();
+
+async function authentication(request, response, next) {
+    const bearerHeader = request.headers['authorization'];
+    if (typeof bearerHeader !== 'undefined') {
+        const bearerToken = bearerHeader.split(' ')[1]
+        try {
+            const authData = jwt.verify(bearerToken, config.secret);
+            request.authorization = authData;
+            request.token = bearerToken;
+            await userController.getUserByEmail(authData.user.email);
```

Comment:
You see, here is the only place where you did the proper getUserByEmail check. This middleware is copied like 5 times across multiple times, and you should fix all of them to include this missing check. Remember to move this check to the middleware implementation in the authentication file.

DRY